include/boost/json/detail/impl/format.ipp

```diff
@@ -111,10 +111,44 @@ format_int64(
 
 unsigned
 format_double(
-    char* dest, double d) noexcept
+    char* dest, double d, const serialize_options & opt) noexcept
 {
-    return static_cast<int>(
-        ryu::d2s_buffered_n(d, dest));
+    if (std::isnan(d))
+    {
+        std::copy(opt.nan.begin(), opt.nan.end(), dest);
```

Comment:
did you mean
```
opt.copy( dest, 0);
```
?

Reply:
I did not mean that, because I don't see why.

Reply:
So we don't have to include `<algorithm>`, which you also forgot